internal/pkg/aws/sqs/sqs.go

```diff
@@ -0,0 +1,165 @@
+}
+
+// QueueAttributes contains a Cloudwatch queue attributes.
+type QueueAttributes struct {
+	Name                                  string         `json:"name"`
+	ARN                                   string         `json:"arn"`
+	ApproximateNumberOfMessages           int            `json:"messages"`
+	ApproximateNumberOfMessagesDelayed    int            `json:"messagesDelayed"`
+	ApproximateNumberOfMessagesNotVisible int            `json:"messagesNotVisible"`
+	RedrivePolicy                         *RedrivePolicy `json:"redrivePolicy"`
+}
+
+// RedrivePolicy contains the fields needed to unmarshal a SQS Queue redrive policy.
+type RedrivePolicy struct {
+	DeadLetterTargetArn string `json:"deadLetterTargetArn"`
+	MaxReceiveCount     int    `json:"maxReceiveCount"`
+}
+
+// New returns a SQS struct configured against the input session.
+func New(s *session.Session) *SQS {
+	return &SQS{
+		client:   sqs.New(s),
+		rgClient: rg.New(s),
+	}
+}
+
+// QueueAttributes returns relevant queue attributes, such as messages and redrive policy
+func (sq *SQS) QueueAttributes(queues []*Queue) ([]QueueAttributes, error) {
+	if len(queues) == 0 {
+		return nil, nil
+	}
+
+	var queueAtts []QueueAttributes
+	for _, queue := range queues {
+		attOut, err := sq.client.GetQueueAttributes(
+			&sqs.GetQueueAttributesInput{
+				QueueUrl: &queue.url,
+				AttributeNames: []*string{
+					aws.String("ApproximateNumberOfMessages"),
+					aws.String("ApproximateNumberOfMessagesDelayed"),
+					aws.String("ApproximateNumberOfMessagesNotVisible"),
+					aws.String("QueueArn"),
+					aws.String("RedrivePolicy"),
+				},
+			})
+		if err != nil {
+			return nil, fmt.Errorf("list SQS queue attributes: %w", err)
+		}
+		redrive, err := parseDLQPolicy(aws.StringValue(attOut.Attributes["RedrivePolicy"]))
+		if err != nil {
+			return nil, fmt.Errorf("list SQS queue redrive policy: %w", err)
+		}
+		messages, err := strconv.Atoi(aws.StringValue(attOut.Attributes["ApproximateNumberOfMessages"]))
+		if err != nil {
+			return nil, err
+		}
+		messagesDelayed, err := strconv.Atoi(aws.StringValue(attOut.Attributes["ApproximateNumberOfMessagesDelayed"]))
+		if err != nil {
+			return nil, err
+		}
+		messagesNotVisible, err := strconv.Atoi(aws.StringValue(attOut.Attributes["ApproximateNumberOfMessagesNotVisible"]))
+		if err != nil {
+			return nil, err
+		}
+
+		queueAtts = append(queueAtts, QueueAttributes{
+			Name:                                  queue.name,
+			ARN:                                   aws.StringValue(attOut.Attributes["QueueArn"]),
+			RedrivePolicy:                         redrive,
+			ApproximateNumberOfMessages:           messages,
+			ApproximateNumberOfMessagesDelayed:    messagesDelayed,
+			ApproximateNumberOfMessagesNotVisible: messagesNotVisible,
+		})
+
+	}
+
+	return queueAtts, nil
+}
+
+// SQSQueues lists out SQS queues deployed by copilot for a given service.
+func (sq *SQS) SQSQueues(appName, envName, svcName string) ([]QueueAttributes, error) {
```

Comment:
For example in the context of the `aws/sqs` package, this method would allow a `tags map[string]string` so that we can fetch queues using tags and then in a client package we would call:
```go
queues, err := client.SQSQueues(map[string]string{
  copilotAppTag: appName,
  copilotEnvTag: envName,
  copilotSvcTag: svcName,
})
```